go-controller/pkg/ovn/namespace.go

```diff
@@ -30,11 +30,8 @@ func (oc *Controller) syncNamespaces(namespaces []interface{}) {
 
 	err := oc.addressSetFactory.ForEachAddressSet(func(hashedAddrSetName, namespaceName, nameSuffix string) {
 		if nameSuffix == "" && !expectedNs[namespaceName] {
-			// delete the address sets for this policy from OVN
-			_, stderr, err := util.RunOVNNbctl("--if-exists", "destroy", "address_set", hashedAddrSetName)
-			if err != nil {
-				klog.Errorf("failed to destroy address set %q, stderr: %q, (%v)",
-					hashedAddrSetName, stderr, err)
+			if err := oc.addressSetFactory.DestroyAddressSetInBackingStore(hashedAddrSetName); err != nil {
```

Comment:
`DestroyAddressSetInBackingStore` takes a name, not a hashed name. I think you can just change `ForEachAddressSet` to return the unhashed name instead of the hashed one

Reply:
@dcbw did you miss updating this one?

Reply:
@trozet try now